sat/cli/bootsys/platform.py

```diff
@@ -484,7 +485,20 @@ def do_recreate_cronjobs(_):
     """Recreate cronjobs that are not being scheduled on time."""
     try:
         batch_api = load_kube_api(api_cls=BatchV1Api)
-        recreate_namespaced_stuck_cronjobs(batch_api, 'services')
+        # Adding a retry
+        max_retries = 3
+        retries = 0
+        while retries < max_retries:
+            try:
+                recreate_namespaced_stuck_cronjobs(batch_api, 'services')
+                break  # Exit the loop if successful
+            except SSLEOFError as e:
```

Comment:
This isn't actually the type of exception that is ultimately raised to this point if you look closely at the traceback in CASMTRIAGE-6300.

Reply:
return ssl_context.wrap_socket(sock)
  File "/usr/lib/python3.9/ssl.py", line 501, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.9/ssl.py", line 1074, in _create
    self.do_handshake()
  File "/usr/lib/python3.9/ssl.py", line 1343, in do_handshake
    self._sslobj.do_handshake()
**ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:1129)**

Had a look at the traceback, your referring to this specific error highlighted instead of having a broad way. Hope the changes are fine

Reply:
The full traceback is as follows:

```
Traceback (most recent call last):
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 714, in urlopen
    httplib_response = self._make_request(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 403, in _make_request
    self._validate_conn(conn)
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1053, in _validate_conn
    conn.connect()
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connection.py", line 419, in connect
    self.sock = ssl_wrap_socket(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 453, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls)
  File "/sat/venv/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 495, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock)
  File "/usr/lib/python3.9/ssl.py", line 501, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.9/ssl.py", line 1074, in _create
    self.do_handshake()
  File "/usr/lib/python3.9/ssl.py", line 1343, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:1129)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sat/venv/bin/sat", line 8, in <module>
    sys.exit(main())
  File "/sat/venv/lib/python3.9/site-packages/sat/main.py", line 122, in main
    subcommand(args)
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootsys/main.py", line 96, in do_bootsys
    stage(args)
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootsys/platform.py", line 670, in do_platform_start
    do_platform_action(args, 'start')
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootsys/platform.py", line 628, in do_platform_action
    step.action(ncn_groups)
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootsys/platform.py", line 487, in do_recreate_cronjobs
    recreate_namespaced_stuck_cronjobs(batch_api, 'services')
  File "/sat/venv/lib/python3.9/site-packages/sat/cronjob.py", line 109, in recreate_namespaced_stuck_cronjobs
    for cronjob in batch_api.list_namespaced_cron_job(namespace).items:
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/api/batch_v1_api.py", line 1437, in list_namespaced_cron_job
    return self.list_namespaced_cron_job_with_http_info(namespace, **kwargs)  # noqa: E501
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/api/batch_v1_api.py", line 1552, in list_namespaced_cron_job_with_http_info
    return self.api_client.call_api(
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 348, in call_api
    return self.__call_api(resource_path, method,
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 180, in __call_api
    response_data = self.request(
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 373, in request
    return self.rest_client.GET(url,
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/rest.py", line 240, in GET
    return self.request("GET", url,
  File "/sat/venv/lib/python3.9/site-packages/kubernetes/client/rest.py", line 213, in request
    r = self.pool_manager.request(method, url,
  File "/sat/venv/lib/python3.9/site-packages/urllib3/request.py", line 77, in request
    return self.request_encode_url(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/request.py", line 99, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/sat/venv/lib/python3.9/site-packages/urllib3/poolmanager.py", line 376, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 826, in urlopen
    return self.urlopen(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 826, in urlopen
    return self.urlopen(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 826, in urlopen
    return self.urlopen(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 798, in urlopen
    retries = retries.increment(
  File "/sat/venv/lib/python3.9/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='10.252.1.2', port=6442): Max retries exceeded with url: /apis/batch/v1/namespaces/services/cronjobs (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1129)')))
```

Note in particular the line that says this that separates the two call stacks:

```
During handling of the above exception, another exception occurred:
```

This means that while urllib3.connectionpool was handling the `SSLEOFError`, it ended up raising another exception of type `urllib3.exceptions.MaxRetryError`. This is the exception that the `sat` code actually sees. This `urllib3.exceptions.MaxRetryError` is a subclass of a more general `urllib3.exceptions.RequestError`. It might make sense to catch the slightly more general `RequestError`.